kernel/src/fs/utils/inode.rs

```diff
@@ -434,6 +495,45 @@ pub trait Inode: Any + Sync + Send {
     fn extension(&self) -> Option<&Extension> {
         None
     }
+
+    ///  Used to check for read/write/execute permissions on a file.
+    ///
+    ///  We use "fsuid" for this, letting us set arbitrary permissions
+    ///  for filesystem access without changing the "normal" uids which
+    ///  are used for other things..
+    fn check_permission(&self, mut req: RequestPermission) -> Result<()> {
```

Comment:
The following code can be improved for readability:
1. Try not to put all logic inside one large `if let { ... }` block, return ealy instead:
```rust
let posix_thread = match current_thread!().as_posix_thread() {
        Some(thread) => thread,
        None => return Ok(()),
    };
/* Code omitted */
```
2. For better debugging, use `return_errno_with_message!()`, e.g., `return_errno_with_message!(Errno::EACCES, "owner permission check failed")`.

Reply:
Done